### ISSUE TITLE
Re-sampling between different kHz / Choosing the sample rate at recording

### ISSUE THREAD

**Issue by RustoMCSpit** [FossifyOrg/Voice-Recorder#33]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Voice-Recorder/issues) or [closed](https://github.com/FossifyOrg/Voice-Recorder/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Voice-Recorder/discussions) or [closed](https://github.com/FossifyOrg/Voice-Recorder/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request. (Potentially two, needs triage. These are so linked that it's hard to tell if it's one)
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Voice-Recorder/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

There are different kHz amounts: 44.1 kHz, 48kHz etc.
Allowing a user to choose the sample rate and resample audio is thus needed.

### Why do you want this feature?

Some programs take different amounts, different file sizes, etc.

### Additional information

https://github.com/FossifyOrg/Voice-Recorder/issues/32 related to this

**Comment by SilverPuppy** [comment]
Resampling is not needed. The ability to choose the sample rate for the new recording is important. The fact that it is not possible stops me from resolving an issue with Opus recording manually. The default sample rate is obviously 44,100 which is unsupported by Opus, so it takes the data, but marks it as 48,000 which results in slightly high pitched playback. If I could manually select 24,000 or 48,000 which are supported, there would be no issue.

**Comment by HT-7** [comment]
Agreed. Both sampling rate and bitrate please. And while you're at it, please add an option for OGG vorbis.

ASR by NLL apps had manual sampling rate and bitrate a decade ago. I would use ASR but because "benevolent" Google has killed all 32-bit apps in Android 14, I can forget about that.

**Comment by SilverPuppy** [comment]
The recent change of the default sampling rate from 44,100 to 48,000 resolves the Opus incompatibility, but having the ability to specify one would be valuable at times. I still believe this needs to be implemented. I think the ideal would be a dropdown with common rates, perhaps

8000  (VoIP)
22050
24000
44100 (CD)
48000 (DVD)
Custom

Alternatively, display a line in settings "Sample Rate: [box displaying number of Hz] and a toggle switch below it "Customize sample rate (may cause errors)" which makes the box editable. Switch off, the box should be grayed and display the default of 48000Hz.

Honestly, anyone who wishes to change the sample rate from the default probably knows what rates to use and doesn't need a drop down menu.

**Comment by naveensingh** [comment]
@SilverPuppy https://github.com/FossifyOrg/Voice-Recorder/issues/new?template=feature_request.yml ;)

**Comment by naveensingh** [comment]
> I still believe this needs to be implemented.

It will be added along with other options commonly found in third party voice recorder apps.
